### ISSUE TITLE
Support WebSocket Endpoints

### ISSUE THREAD

**Issue by matanpramp** [a85/POSTMan-Chrome-Extension#4009]
Websocket support, even the most basic (no tests, code generation, collections), would be very useful.

Basic support IMO would be: 
- the ability to establish connections
- read incoming messages
- send outgoing messages
- maintaining support for existing Postman features such as headers, cookies, viewing and editing the messages

Considering the nature of many modern, interactive web applications, web sockets are becoming a basic requirement for many backend automation tasks.

**Comment by SamvelRaja** [comment]
Thanks for your feedback. I am marking this a feature request. I cannot promise an ETA now.
We will update this thread when we pick this feature.

**Comment by binarymist** [comment]
How's progress?

**Comment by omar-haris** [comment]
Any news ?

**Comment by ikushnir** [comment]
Looking forward to have this feature !

**Comment by loveexception** [comment]
i 'm  love this feature .

**Comment by webcomsystemspty** [comment]
We need this feature

**Comment by supermacro** [comment]
joining the notification train

**Comment by cosad3s** [comment]
Same.

**Comment by calvincs** [comment]
ditto

**Comment by byronvoorbach** [comment]
+1

**Comment by jreisam** [comment]
@@ nothing?

**Comment by xukaifu** [comment]
this feature is awesome

**Comment by kienbui1995** [comment]
This feature is needed.
I think that many users want to it attach the app.
https://stackoverflow.com/questions/42269075/websocket-connections-with-postman/43754722#43754722

**Comment by maddestructor** [comment]
This feature would be a game changer to our team!

**Comment by wave100** [comment]
It'd be awesome if this became a feature!

**Comment by Whyner** [comment]
+1

**Comment by fborie** [comment]
+1

**Comment by duart38** [comment]
+9999 I've been using this as i work on some other projects. [link](https://github.com/duart38/socketer)

**Comment by tqwewe** [comment]
+69

**Comment by MaxVynohradov** [comment]
+1

**Comment by cisolarix** [comment]
+10086

**Comment by fredrickreuben** [comment]
+1

**Comment by me89** [comment]
maybe we should scream here:
https://community.getpostman.com/c/feature-requests

**Comment by me89** [comment]
maybe we should ask here:
https://community.getpostman.com/c/feature-requests

**Comment by supermacro** [comment]
Also, see roadmap (which sadly doesn't have any mention of websockets). 

https://trello.com/b/4N7PnHAz/postman-roadmap-for-developers

Would be nice if the postman team would communicate why websockets seem to be such a low priority.

**Comment by libdefi** [comment]
+1

**Comment by 1990prashant** [comment]
+1

**Comment by 3rdstage** [comment]
+1

**Comment by supermacro** [comment]
Hey can we stop with the +1's?

I feel partly to blame for this. But it's not adding to the conversation and I'm sure the postman team doesn't appreciate it.

Also, I'm sure everyone else in this thread is getting annoyed of seeing the blue notification bubble on github light up because of a meaningless "+1" comment.

**Comment by mindfocus** [comment]
Postman Makes API Development Simple.
Websocket support CAN makes API Development Simple.
so why not?

**Comment by Priultimus** [comment]
+1

**Comment by mashimaroa15** [comment]
> Hey can we stop with the +1's?
> 
> I feel partly to blame for this. But it's not adding to the conversation and I'm sure the postman team doesn't appreciate it.
> 
> Also, I'm sure everyone else in this thread is getting annoyed of seeing the blue notification bubble on github light up because of a meaningless "+1" comment.

Without the number of +1, how can Postman team know which feature that is important to their fans? Although it depends also many other aspects like workforce, feasibility regarding existing code structure... In short, it's not meaningless for me

+1 for this feature

**Comment by jhpratt** [comment]
@mashimaroa15 Use the "reactions" feature, where you can add a thumbs up on the original post. You don't need a separate comment.

**Comment by jentyk** [comment]
+1

**Comment by me89** [comment]
What a rebellion here. Stand up and fight for this feature :D

**Comment by a85** [comment]
Folks, We use websockets as well and need Postman for Websockets to test Postman. I don’t have an update right now but we are building plans for supporting this. 

Also, we are hiring and expanding fast. So if any of you are interested in helping us build this, let me know! ;)

**Comment by danielmogin** [comment]
+1

**Comment by abelzoltan** [comment]
+1

**Comment by kelsonic** [comment]
+1

**Comment by dakleine** [comment]
+1

**Comment by CodeFuentes** [comment]
+1 its a shame having to switch applications when testing apis/ws, would love to do it all in postman.

**Comment by zenonwch** [comment]
+1

**Comment by DarkHonin** [comment]
This thread is a beautiful mess to my sleep deprived eyes at 5:15 am trying to find a way to test these whisps of the web

**Comment by emmanuellmota** [comment]
+1

**Comment by dineshvg** [comment]
This is needed.

**Comment by theomarshanti** [comment]
+1

**Comment by smartnav** [comment]
+1

**Comment by seitbekir** [comment]
Is there is a plugin support for Postman and the plugin that implements this feature?

**Comment by sharypovandrey** [comment]
Postman, we love you! Give us a WS testing tool, please

**Comment by zl7261** [comment]
+10086

**Comment by galanoff** [comment]
+2

**Comment by gjsterenborg** [comment]
+1

**Comment by AsafChetritContentSquare** [comment]
+1

**Comment by jrhuerga** [comment]
I would like that feature as well. This is a command line alternative: https://www.npmjs.com/package/wscat

**Comment by adriendengreville** [comment]
I'm craving for this websocket feature, +1

**Comment by inigo333** [comment]
still nothing?

**Comment by Mojicode** [comment]
It will be very wonderful if the postman with websocket

**Comment by IngCr3at1on** [comment]
It's official, I can no longer remain subscribed to this issue waiting for an actual update because people are too busy posting the same things over and over again in blatant disregard for both repo maintainer wishes as well as GitHub general community suggestions.

I really hope that the postman team comes up with a decent way to announce when this feature is complete for those of us who are no-longer watching this issue.

/unsub

**Comment by FJEANNOT** [comment]
+1

**Comment by filipemacedo** [comment]
+1

**Comment by im4eversmrt** [comment]
+1

**Comment by douglas444** [comment]
+1

**Comment by piavgh** [comment]
+1

**Comment by a85** [comment]
Everyone, thank you for your patience. Unfortunately, no ETA on this yet. We are scaling up the team to support other protocols and APIs. We use Websockets ourselves as our real-time collaboration service uses websockets. Separately, we are hiring across the globe for our engineering team. If you want to help us implement this, please check out our job page here: https://www.getpostman.com/jobs/

**Comment by dverbovyi** [comment]
no updates or even ETA for 1.5 years( it's sad

suckman

**Comment by YJinHai** [comment]
+1
Very, very sad

**Comment by webhacking** [comment]
-1

**Comment by gepopa** [comment]
Beer for all the team when postman will support websockets :D

**Comment by aCodeAlchemist** [comment]
**WOW!** When searching for socket.io testing tools on chrome webstore I found this precise tool called [firecamp](https://chrome.google.com/webstore/detail/firecamp-a-campsite-for-d/eajaahbjpnhghjcdaclbkeamlkepinbl?hl=en).. check this out guys.. 

![Capture](https://user-images.githubusercontent.com/8849256/58928427-0d419700-8770-11e9-8b7d-12a9c75f6188.PNG)

also it supports websockets and bunch of other things too. I have been playing with that for couple weeks now.. I can test and debug realtime events pretty quick. Kudos to developers. ✌

**Comment by sotanodroid** [comment]
Need this one

**Comment by lefuturiste** [comment]
+1

**Comment by fe19880924** [comment]
+1

**Comment by aaronn** [comment]
+1

**Comment by exceptionplayer** [comment]
Need this.

**Comment by trickstival-logcomex** [comment]
+1

**Comment by SLYJason** [comment]
no updates?

**Comment by miguel550** [comment]
+1

**Comment by btargac** [comment]
neeed websocket on postmaan

**Comment by Benoit1980** [comment]
yes please web sockets for postman, found out today you cannot use it with WebSocket, sadly.

**Comment by tiny1990** [comment]
+1

**Comment by BillRozy** [comment]
+1

**Comment by yanxinyuan** [comment]
+1

**Comment by eabovsky** [comment]
+1

**Comment by fapostolo** [comment]
lol one more bites the dust

**Comment by VishnuPriyaChandraSekar** [comment]
+1

**Comment by joshua-jingu-lee** [comment]
+1

**Comment by thewilli** [comment]
There was once a platform called GitHub. From time to time somesomeone opened a topic for discussion. And if it happenend that this topc was of high interest, many people shouted `me too`. While it was at first helpful for the recipient to see this was generally recognized, it soon started to get so annoying, that the topic's recipient and sometimes even the topic's creator lost interesent. So GitHub introduced [reactions](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to allow for showing interest, acceptence or denial to a particular topic - without filling up inboxes with useless `+1` notifications, and their occasional reactions (like this one).

So please, @joshua-jingu-lee, @VishnuPriyaChandraSekar, @eabovsky, @yanxinyuan, @BillRozy and all who are just in this moment considering to add a `+1` comment. Just do what more than 800 (!!) people already did: Use the reactions buttons instead of adding useless comments. Save your time, our nerves and the climate by not requring the power to transmit hundreds of `+1` notifcations to inboxes all over the word, so we can live happily ever after.

**Comment by dewbjorn** [comment]
> There was once a platform called GitHub. From time to time somesomeone opened a topic for discussion. And if it happenend that this topc was of high interest, many people shouted `me too`. While it was at first helpful for the recipient to see this was generally recognized, it soon started to get so annoying, that the topic's recipient and sometimes even the topic's creator lost interesent. So GitHub introduced [reactions](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to allow for showing interest, acceptence or denial to a particular topic - without filling up inboxes with useless `+1` notifications, and their occasional reactions (like this one).
> 
> So please, @joshua-jingu-lee, @VishnuPriyaChandraSekar, @eabovsky, @yanxinyuan, @BillRozy and all who are just in this moment considering to add a `+1` comment. Just do what more than 800 (!!) people already did: Use the reactions buttons instead of adding useless comments. Save your time, our nerves and the climate by not requring the power to transmit hundreds of `+1` notifcations to inboxes all over the word, so we can live happily ever after.

+1

**Comment by kkojot** [comment]
There is an Apic's chrome extension https://chrome.google.com/webstore/detail/apic-complete-api-solutio/ggnhohnkfcpcanfekomdkjffnfcjnjam in which you can test your API with a native WebSocket connection or over STOMP. I know it's a beta version, but for WebSocket, it's still better than Postman...

**Comment by szsascha** [comment]
Any news for websocket support in postman?

**Comment by xtianus79** [comment]
yea +1 so many people are doing it why is the struggle real?  lol. so this is my setup. i make a post request with postman while listening to a socket through another program. come on man!!! lol

**Comment by siredwin** [comment]
I am a loyal customer of Postman for documenting HTTP methods using restful API. But at the moment, I can no longer wait for a WebSocket client. I am creating my own web-based Websocket Client and I will share it soon. The chrome extensions are just not my thing.

**Comment by szsascha** [comment]
@siredwin Sounds really nice! Is it standalone or integrated into Postman?

Isn't Postman open source? Maybe we as a community can contribute some WS functionality?

**Comment by siredwin** [comment]
@SaschaS93, I don't think it is open source. But I am doing web first then stand-alone later. I can integrate to it at some point if they accept such a thing.

**Comment by xtianus79** [comment]
@SaschaS93 socket wrench and firebase which is a chrome extension are really good

**Comment by sunergeo** [comment]
badly needed

**Comment by odwrotnie** [comment]
+1

**Comment by MikeGJennings** [comment]
+1 -- badly needed

**Comment by Sunbreak** [comment]
Firefox just release their own WebSocket Inspector: https://hacks.mozilla.org/2019/10/firefoxs-new-websocket-inspector/

**Comment by DruidKuma** [comment]
Needed as hell. Can't use Postman because of absence of web socket support

**Comment by spyro2000** [comment]
So, this is basically dead...?

**Comment by kelsonic** [comment]
2 years later... 😭

**Comment by nixizi** [comment]
Any news?

**Comment by BerkeBelgin** [comment]
+1

**Comment by dlandis** [comment]
https://postwoman.io/realtime

**Comment by scodeman** [comment]
+1

**Comment by rockitseller** [comment]
+1

**Comment by muriyx** [comment]
+1

**Comment by hsa-embedded** [comment]
+1

**Comment by dmutende** [comment]
+1

**Comment by H6LS1S** [comment]
+1

**Comment by fgustovo** [comment]
After years we are still waiting for you with our tears :(

**Comment by jonathan-s** [comment]
For people reading this issue, just use firecamp -> https://firecamp.io/ it supports websockets.

**Comment by zeljkot** [comment]
> For people reading this issue, just use firecamp -> https://firecamp.io/ it supports websockets.

They are not very clear about the pricing:

> Pricing
> During the Beta release period, it is free to use the product.
> 
> We reserve the right to change this in the future.> 

If I understood correctly, beta is free forever.

**Comment by elliolir** [comment]
still no support 😔😢

**Comment by unel** [comment]
some news?..

**Comment by dreamleilei** [comment]
need this feature

**Comment by moonkop** [comment]
still waiting?

**Comment by sneighborsparamount** [comment]
+1

**Comment by elMestre** [comment]
> still waiting?

Well, I solved my project years ago and... well, not sure if at the end will be done, or not.... but at this point I'm subscribed to this request only for the laughs. 

(Anyway... it reeeally would be a nice feature)

**Comment by sneighborsparamount** [comment]
Yeah thought I would put the +1 there just to keep annoying them…

**Comment by dewbjorn** [comment]
> still waiting?

I'm just here for the +1's

**Comment by elMestre** [comment]
I would have paid for it, unfortunately, also not mentioned on the <a href="https://blog.getpostman.com/2020/01/06/announcing-updated-postman-plans-and-pricing"> paid plans</a>
I can understand how they don't want to deal with WS: incoming connections, dealing with it opened... Lot more things to deal. 
Well... ¯\_(ツ)_/¯

Anyway, you can always connect and redirect it yourself via express or similar. Perhaps a nice weekend project... if this comment attracts smileys I will think about it :P

**Comment by siredwin** [comment]
At this rate, I will get my own Web-based WebSocket client to the market in about 1-2 months. I can understand why they don't want to do it. It is a pain to put together a WebSocket client that actually allows you to use headers. 99% of all clients out there are Javascript based and not very good. Also, many people don't use WebSockets.

**Comment by elMestre** [comment]
Sorry for my previous comment: late night bravata. 0:30AM here. I was thinking on some shabby thing. Kinda ‘works in my system’ and if lucky throws errors trough CLI

You are right on both things. Is not a weekend project and is a pain to deal properly.

**Comment by siredwin** [comment]
@elMestre, there are a few out there like "FireCamp", etc. But you won't be able to test authorization headers with them. That is actually why I built my own.

**Comment by Eden-Harris** [comment]
####  `websocket` API and graphql's `subscription` can not  use,  It ’s already 2020, and I ’m looking forward to supporting websocket.

**Comment by scarbo87** [comment]
Any news?

**Comment by JimLynchCodes** [comment]
+1 😁

**Comment by ArtashMardoyan** [comment]
any news?

**Comment by wh1337** [comment]
+1 👍

**Comment by Venpelmen** [comment]
Any news?

**Comment by DrYSG** [comment]
Postwoman is only for internet sites, we need PostMan for internal graphql subscribe services. If not here, then when Insomnia gets this feature, we will use insomnia over postman

**Comment by roovenier** [comment]
Any news?

**Comment by petyappetrov** [comment]
?

**Comment by mlaass** [comment]
do it now!

**Comment by guy-compie** [comment]
+1 !

**Comment by owenashurst** [comment]
Also interested in this functionality. Would be super useful!

**Comment by jcorvera** [comment]
This feature, is importante :( !!

**Comment by Saqib92** [comment]
https://github.com/liyasthomas/postwoman
it supports Socket.io

**Comment by ghost** [comment]
> https://github.com/liyasthomas/postwoman
> it supports Socket.io

See this Postman team? Ignoring users is how software becomes obsolete.

**Comment by alanhe421** [comment]
> > https://github.com/liyasthomas/postwoman
> > it supports Socket.io
> 
> See this Postman team? Ignoring users is how software becomes obsolete.

It‘s time to give up postman.

Choose postwoman

**Comment by SeppPenner** [comment]
Would be a nice feature 👍

**Comment by nuelsoft** [comment]
Still nothing. Do something Please

**Comment by SeppPenner** [comment]
Easiest way in my opinion. Use Chrome debugging console (F12) with something like this:

```JavaScript
ws = new WebSocket("ws://localhost:8080/ws");
ws.onmessage = function (e) {
    console.log(e.data);
};

ws.onopen = function() {
    ws.send("SomeMessage");
}
```

**Comment by aangrisani** [comment]
+1

**Comment by dbhat6** [comment]
+1
Please add this into Postman!

**Comment by the-mars-rover** [comment]
+1

**Comment by quangtd95** [comment]
+1

**Comment by starkast1c** [comment]
+1

**Comment by GioPan04** [comment]
This comment is useless but +1

**Comment by zionitcorp** [comment]
+1

**Comment by nicorodrigues** [comment]
+1

**Comment by SergioBarbero** [comment]
+1

**Comment by matibat** [comment]
It woud be great!

**Comment by nurgasemetey** [comment]
+1

**Comment by veaceslav3** [comment]
Check out [https://gosandy.io/](https://gosandy.io/)
- Connect to any WebSocket, RSocket server
- Save request messages for later re-use
- Share configuration with your mates

![image](https://user-images.githubusercontent.com/64411284/85752706-4f8b9c80-b714-11ea-9600-9167e35edddb.png)

**Comment by yarfik** [comment]
+1

**Comment by ilyaskarim** [comment]
People are waiting and postmanlabs is ignoring all of them. This is ridiculous.

**Comment by alexmayerlab** [comment]
+1

**Comment by MaximusFT** [comment]
+1

**Comment by GioPan04** [comment]
Plz stop commenting with "+1", I'm afraid of resiving useless emails 😂. We all know that this feature will be so useful but if you want to give a "+1" simply react to the first comment 🙂. Thanks and have all you a great day.

**Comment by coderrajput** [comment]
please add this socket feature

**Comment by Saqib92** [comment]
> Plz stop commenting with "+1", I'm afraid of resiving useless emails 😂. We all know that this feature will be so useful but if you want to give a "+1" simply react to the first comment 🙂. Thanks and have all you a great day.

I agree. +1

**Comment by Moizsohail** [comment]
+1

**Comment by thesteve0** [comment]
+1 this way I can do all my REST work in Postman rather than having to do some in postman and then websockets in FF developer console

**Comment by wallevy** [comment]
+1

**Comment by a2mz** [comment]
+1

**Comment by douyoung** [comment]
+1

**Comment by wilfred-asomanii** [comment]
+1

**Comment by mohammedgqudah** [comment]
can yall stop commenting "+1" "this is useful".
just like it and don't do anything else. thanks.

**Comment by MikeGJennings** [comment]
Maybe we wouldn't feel the need to add the noise if the Postman team would just drop in and let us know what they're thinking.

**Comment by larryding1981** [comment]
I am eagerly waiting for it.

**Comment by ilyaskarim** [comment]
This issue should get award for the ignored issues of Github.

**Comment by mohammedgqudah** [comment]
@SamvelRaja any updates? are you going to implement this feature cuz we are all waiting.

**Comment by Niach** [comment]
+1 !!!!

**Comment by arlemi** [comment]
Hey everyone,

We've given our point of view on this previously, admittedly it's not that easy to find because of all the +1s so here it is:  
> https://github.com/postmanlabs/postman-app-support/issues/4009#issuecomment-432861109: Folks, We use websockets as well and need Postman for Websockets to test Postman. I don’t have an update right now but we are building plans for supporting this.
Also, we are hiring and expanding fast. So if any of you are interested in helping us build this, let me know! ;) 

> https://github.com/postmanlabs/postman-app-support/issues/4009#issuecomment-489468553: Everyone, thank you for your patience. Unfortunately, no ETA on this yet. We are scaling up the team to support other protocols and APIs. We use Websockets ourselves as our real-time collaboration service uses websockets. Separately, we are hiring across the globe for our engineering team. If you want to help us implement this, please check out our job page here: https://www.getpostman.com/jobs/

As said before this is something we want to build, but we do not have an ETA yet. 
We see and read all your comments, be sure you'll be the first ones updated when there's something new on this front (and/or if we need beta testers). 🙂 

That said: 
- If you want to stay up to date, use the **Subscribe** button on the right hand side
- If you'd like to show support for the feature, add a 👍 on the original post to avoid sending notifications to everyone in the thread

**Comment by westofer** [comment]
What happened to the creator of the issue though? @matanpramp 
Are you still alive? Do you still work on web development?  Hope you are well

Edit: you might want to check https://postwoman.io which supports web sockets

**Comment by SwapnilSoni1999** [comment]
2020 and here we go

**Comment by theGeekyLad** [comment]
Hoping to see a beta release at the least, with the basic HTTP handshake and required upgrade headers by the spec. What do ya'll think?

**Comment by rockycookie** [comment]
+1

**Comment by connorlaw2151** [comment]
+1

**Comment by Friedrich42** [comment]
+1

**Comment by Friedrich42** [comment]
I support this feature

**Comment by spacepirate0001** [comment]
Need this feature! Thanks for the great work.

**Comment by hanayashiki** [comment]
its almost 2021

**Comment by netokey** [comment]
+1

**Comment by tusharsadhwani** [comment]
bump

**Comment by 4nif** [comment]
any plan for this?

**Comment by dzybkin** [comment]
hello

**Comment by ilyaskarim** [comment]
Hi

**Comment by szsascha** [comment]
The announcement of websocket support in Postman would be the only thing that could save 2020.

**Comment by fmvilas** [comment]
Well, not exactly an announcement of WebSockets support in Postman but last week we [announced a partnership with AsyncAPI](https://blog.postman.com/asyncapi-joins-forces-with-postman-future-of-apis/). So WebSockets will only be the beginning of a richer multi-protocol support 😊

In the initial comment, @matanpramp outlined a few features they'd love to have:

* the ability to establish connections
* read incoming messages
* send outgoing messages
* maintaining support for existing Postman features such as headers, cookies, viewing and editing the messages

Is there anything else you think would be useful for you? Let's get this done! 🚀

**Comment by szsascha** [comment]
> Well, not exactly an announcement of WebSockets support in Postman but last week we [announced a partnership with AsyncAPI](https://blog.postman.com/asyncapi-joins-forces-with-postman-future-of-apis/). So WebSockets will only be the beginning of a richer multi-protocol support 😊
> 
> In the initial comment, @matanpramp outlined a few features they'd love to have:
> 
> * the ability to establish connections
> * read incoming messages
> * send outgoing messages
> * maintaining support for existing Postman features such as headers, cookies, viewing and editing the messages
> 
> Is there anything else you think would be useful for you? Let's get this done! 🚀

Sound‘s great! I didn‘t know AsyncAPI before. It just look‘s awesome! Have been searching years for something like this. AsyncAPI in Postman is the greatest thing in the future of API development I could imagine. This will be a great asset in terms of productivity and standardization in the field of API development.

In my opinion one of the most important features in Postman is the testing capability. This feature in combination with websocket support would be a huge gamechanger. Looking forward for automated websocket testing with Newman.

You definitely made my 2020 with this post. Looking forward for 2021 for more news regarding this.

Thanks a lot for sharing this news!

**Comment by 4nif** [comment]
any timeline for websocket support @fmvilas ?

**Comment by lequocdien** [comment]
+1

**Comment by higorrezend** [comment]
Looking forward to this feature. Any date when it will be available?

**Comment by ilyaskarim** [comment]
No, postman put this issue as spam.

**Comment by jdukleth** [comment]
+1

**Comment by deefdragon** [comment]
If nothing else, would it be possible to get [ws](https://www.npmjs.com/package/ws) or some other client library added to the available NodeJS libs if nothing else?
Also: +1

**Comment by siredwin** [comment]
I have a fully functioning websoket tester if you need it. Websokets end to end

**Comment by Churakovmike** [comment]
+1

**Comment by Aziiiz** [comment]
+1

**Comment by tonghoang196** [comment]
+1

**Comment by Odirb** [comment]
+1

**Comment by abarazal** [comment]
We really need this

**Comment by iagxferreira** [comment]
+1

**Comment by jdelamare** [comment]
+1

**Comment by aradalvand** [comment]
2021 and still waiting...

**Comment by ghost** [comment]
bumpity bump bump

**Comment by professorlevi** [comment]
> I have a fully functioning websoket tester if you need it. Websokets end to end

Hey, why don't you share it with us

**Comment by AngeloD2022** [comment]
+1

**Comment by vasmancs** [comment]
+1

**Comment by Fliko** [comment]
+1

**Comment by kivals** [comment]
+1

**Comment by ghost** [comment]
Please 🙏

**Comment by pandier** [comment]
+1

**Comment by veve90** [comment]
+1 This would be great!!

**Comment by ViktorBash** [comment]
+1

**Comment by mjhaghparast** [comment]
❤

**Comment by nabil-alhfz** [comment]
++++

**Comment by acebelowzero** [comment]
+1

**Comment by RGlanville** [comment]
+1

**Comment by oznecniV97** [comment]
+1

**Comment by afefer** [comment]
This would also be super useful for testing GraphQL subscriptions.

**Comment by itsknk** [comment]
This would be insanely useful for testing the GraphQL subscriptions.

**Comment by ejoo** [comment]
Let's make a team and start an open-source project

**Comment by itsknk** [comment]
> This would also be super useful for testing GraphQL subscriptions.

Hey, you can use this [tool](https://altair.sirmuel.design/) for working with graphql subscriptions.

**Comment by afefer** [comment]
@shamasis @a85 @vkaegis @SamvelRaja any chance this can be revisited as part of Postman feature planning? It's been more than 3 years since this feature request was submitted - want to bring this back to your radar as there are ton of scenarios where this would be useful. 216 comments and nearly 2000 upvotes hopefully indicates the demand.

**Comment by fmvilas** [comment]
Hey @afefer! Support for WebSockets is currently in development. Can't say more but this is going to be awesome 🙊

**Comment by ernst77** [comment]
need this badly...

**Comment by TylerSustare** [comment]
I think this might make me go back to Postman instead of using Insomnia

**Comment by M4THYOU** [comment]
@fmvilas Thanks for letting us know!

**Comment by laureanray** [comment]
This would be a really neat feature if implemented!

**Comment by SwapnilSoni1999** [comment]
> Hey @afefer! Support for WebSockets is currently in development. Can't say more but this is going to be awesome

Will they select students from gsoc to implement this feature?

**Comment by arlemi** [comment]
@SwapnilSoni1999 The feature is already in development, so probably not! If you're interested about the different ideas we've put out for GSoC, please check our community forum: https://community.postman.com/c/open-technology/gsoc/42

**Comment by SwapnilSoni1999** [comment]
> @SwapnilSoni1999 The feature is already in development, so probably not! If you're interested about the different ideas we've put out for GSoC, please check our community forum: https://community.postman.com/c/open-technology/gsoc/42

Thats great <3 I'm planning for gsoc this year lets see how can I be helpful for this community. Thanks alot

**Comment by lionuncle** [comment]
Any ETA? April 2021?

**Comment by BrettThib** [comment]
Yes, please!

**Comment by Roberto-Calderon** [comment]
+1

**Comment by nazmulmamunh** [comment]
1

**Comment by brdilshod** [comment]
+1

**Comment by skybird23333** [comment]
just click the subscribe button to get notifs

**Comment by GioPan04** [comment]
This issue should be locked until any words from postman team

**Comment by dmartinsid** [comment]
+1

**Comment by dcmdeivid** [comment]
+1

**Comment by jgowdy** [comment]
👍🏻 
+1

**Comment by arlemi** [comment]
Hey everyone! If you're interested WebSocket's support in Postman you might want to join our livestream on [YouTube](https://www.youtube.com/watch?v=Sr-fq0eqsgk) or [Twitch](https://www.twitch.tv/getpostman) at 7PM PST tomorrow, we'll have some news to share! 

And if you're not able to attend live, don't worry, we'll share the news here too. 😉

**Comment by appurva21** [comment]
Hello everyone 👋 !

We are immensely excited to announce that Postman now supports the testing of WebSocket APIs! 🎉

In the latest version of Postman, you can now hit the `New` button in the left sidebar and open a WebSocket Request tab. Check out our [Learning Center documentation](https://learning.postman.com/docs/sending-requests/supported-api-frameworks/websocket/) to know more about how to use Postman for WebSocket APIs

![websocket-connect](https://user-images.githubusercontent.com/24985700/118992666-7729ca00-b9a2-11eb-9419-6db4e0f733fe.gif)


Read more about this on the [Postman Blog](https://blog.postman.com/postman-supports-websocket-apis/) and share your feedback with us in our [discussion forum](https://community.postman.com/t/user-feedback-websocket-support/23879).

Lastly, a big shoutout to our Postman community for providing all the great suggestions here and being patient with us 🙌  !

**Comment by shamasis** [comment]
Thanks everyone for patiently waiting for this. Completing the loop around our [API Platform](https://www.postman.com/api-platform/) constructs was crucial alongside plugging in other protocols.

I would like to extend a line to @matanpramp for reporting this feature request in the most succinct way possible and to @gepopa - the team is excited to someday meet and share 🍻. Not to mention every upvotes we received to understand what we need to build.

Support for subprotocols (Socket.io, etc) + integrating with collections to test, document, share and use AsyncAPI Schema are all coming soon. This is still in beta and our goal is to iterate based on your feedback. Please file feature requests as new issues and we will keep collating them.

**Comment by altruios** [comment]
> Thanks everyone for patiently waiting for this. Completing the loop around our [API Platform](https://www.postman.com/api-platform/) constructs was crucial alongside plugging in other protocols.
> 
> I would like to extend a line to @matanpramp for reporting this feature request in the most succinct way possible and to @gepopa - the team is excited to someday meet and share 🍻. Not to mention every upvotes we received to understand what we need to build.
> 
> Support for subprotocols (Socket.io, etc) + integrating with collections to test, document, share and use AsyncAPI Schema are all coming soon. This is still in beta and our goal is to iterate based on your feedback. Please file feature requests as new issues and we will keep collating them.

okay - so socket.io is not yet supported? (wish I found this post yesterday, lol), any timeline on when socket.io support might be?

**Comment by shamasis** [comment]
> > Thanks everyone for patiently waiting for this. Completing the loop around our [API Platform](https://www.postman.com/api-platform/) constructs was crucial alongside plugging in other protocols.
> > I would like to extend a line to @matanpramp for reporting this feature request in the most succinct way possible and to @gepopa - the team is excited to someday meet and share 🍻. Not to mention every upvotes we received to understand what we need to build.
> > Support for subprotocols (Socket.io, etc) + integrating with collections to test, document, share and use AsyncAPI Schema are all coming soon. This is still in beta and our goal is to iterate based on your feedback. Please file feature requests as new issues and we will keep collating them.
> 
> okay - so socket.io is not yet supported? (wish I found this post yesterday, lol), any timeline on when socket.io support might be?

Socket.io is already supported. @codenirvana is writing a doc around the specifics. However we want to also build a more dedicated UI support for Socket.io so that you do not need to know the underlying workings of Socket.io while using Postman. That extended support is coming up next. We need to iron out some design concepts before we can confidently convey a date.

**Comment by prabhakarj13** [comment]
@Shamasis is there any support of subprotocols: graphql-ws  for websocket

**Comment by codenirvana** [comment]
@prabhakarj13 not yet.

Please follow https://github.com/postmanlabs/postman-app-support/issues/9997 for updates on subprotocols support.

**Comment by shirshak55** [comment]
looks like its not implemented properly. Like I need to go to new and click on websocket? Why not just new tab and check if url has ws:// or wss://? Like how do new user literally discover how to use websockets?

And second we cannot save? Like if there is multiple message if we disconnect and reconnect we lose all the message? I hope this will be fixed once beta is over. But closing this issue without implementing fully might be premature?

**Comment by shirshak55** [comment]
looks like its not implemented properly. Like I need to go to new and click on websocket? Why not just new tab and check if url has ws:// or wss://? Like how do new user literally discover how to use websockets?

And second we cannot save? Like if there is multiple message if we disconnect and reconnect we lose all the message? I hope this will be fixed once beta is over. But closing this issue without implementing fully might be premature?

**Comment by arlemi** [comment]
@shirshak55 You can access previous requests/message history from the History tab (left sidebar). The feature is still in beta and we'll be building on top of the feedback we receive here.

**Comment by shamasis** [comment]
@shirshak55 ... thank you for the feedback. @arlemi answered one of your questions.

In the grand scheme of API lifecycle, there is a lot more to do for WebSockets and the current implementation enables the core constructs as described in this issue. https://github.com/postmanlabs/postman-app-support/issues/4009#issuecomment-845165504

We are tracking all subsequent bugs and feature requests separately to ensure everything is not stuffed in one long running issue for many years! Please use this GH label to see what we have distilled so far https://github.com/postmanlabs/postman-app-support/labels/websocket

I would really love to read more on how you are using Websocket within Postman and the things that we can do when we save a websocket request within a collection.

**Comment by shirshak55** [comment]
Actually, history is not that good tbh. Because after testing one or two endpoint and updating messages there are hundreds of history and I don't know what are the changes. I think it will be improved in future.  Can't we just save the request just like another http get/post etc? 

I understand ws is a different beast but I wished ws would get more focus. The web is no longer just http1.1.

And history doesn't work afaik.

![image](https://user-images.githubusercontent.com/8097377/123286955-19067e80-d52e-11eb-9a5e-39d5035777d8.png)

**Comment by shamasis** [comment]
> Actually, history is not that good tbh. Because after testing one or two endpoint and updating messages there are hundreds of history and I don't know what are the changes. I think it will be improved in future.  Can't we just save the request just like another http get/post etc? 
> 
> 
> 
> I understand ws is a different beast but I wished ws would get more focus. The web is no longer just http1.1.
> 

Keep your eyes on our Twitter and release notes. More on ws will keep coming in more often. Socket.io support and collection integration (in some limited scope though.)

**Comment by meenakshi-dhanani** [comment]
You might be interested in a live stream happening today about how the Postman team built support for web sockets, join us with all your questions - [YouTube](https://www.youtube.com/watch?v=pvucSmRnbXA&feature=youtu.be) or [Twitch](https://www.twitch.tv/getpostman) 8am PT

**Comment by shamasis** [comment]
> > Actually, history is not that good tbh. Because after testing one or two endpoint and updating messages there are hundreds of history and I don't know what are the changes. I think it will be improved in future.  Can't we just save the request just like another http get/post etc? 
> > 
> > 
> > 
> > I understand ws is a different beast but I wished ws would get more focus. The web is no longer just http1.1.
> > 
> 
> Keep your eyes on our Twitter and release notes. More on ws will keep coming in more often. Socket.io support and collection integration (in some limited scope though.)

Socket.IO support is now available on latest Postman - https://blog.postman.com/postman-now-supports-socket-io/

**Comment by vansham-gortnm** [comment]
I am wondering how to emit an event to the `socket.io` based socket server. There appears to be no concise explanation for that and everytime I try to emit anything, the connection closes.

**Comment by codenirvana** [comment]
@vansham-gortnm can you create a [new bug report](https://github.com/postmanlabs/postman-app-support/issues/new/choose) or reaching out to the wider [Postman Community](https://community.getpostman.com/) for help?

Locking this closed thread conversation to avoid unnoticed bug reports and feature requests.

---

Everyone, feel free to raise a feature request and bug report related to WebSocket request here: https://github.com/postmanlabs/postman-app-support/issues/new/choose
